icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -6133,6 +6134,13 @@ void RBBITest::TestBug22636() {
     assertEquals(WHERE, ec, U_ZERO_ERROR);
 }
 
+void RBBITest::TestBug22848() {
+    if (quick || logKnownIssue("ICU-22848", "Very slow case")) { return; }
```

Comment:
I recommend
-  reduce the number of dots in the pattern to the minimum that causes the failure (copy from bug report comment). This will dramatically speed up the test.
-  add an AssertSuccess after creating the bi. So the test will actually fail, and not just hang.
-  re-title the PR to something like "Add test for rbbi rule builder failure."

The test should fail -- when the state table max size is exceeded the ec will return a failure.